c/version.h

```diff
@@ -311,7 +311,7 @@ typedef int tputsputcchar;
 #define MMAP_HEAP
 #define IEEE_DOUBLE
 /* for both iPhone and iPhoneSimulator */
-#if defined(TARGET_OS_IPHONE)
+#if TARGET_OS_IPHONE
```

Comment:
When I build this patch on my machine with clang 16, I get warnings like:

```
gcc -DPORTABLE_BYTECODE -Ipb/boot/pb -Ipb/c -Ic/ -Ipb/zlib -Ipb/lz4/lib -arch arm64 -O2 -Wpointer-arith -Wall -Wextra -Wno-implicit-fallthrough -o pb/c/pb.o -c c/pb.c
In file included from c/pb.c:1:
In file included from c/system.h:29:
c/version.h:314:5: warning: 'TARGET_OS_IPHONE' is not defined, evaluates to 0 [-Wundef-prefix=TARGET_OS_]
  314 | #if TARGET_OS_IPHONE
      |     ^
1 warning generated.
```

Maybe it would be better to go with `#if defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE` instead?

Reply:
Ugh, ok. I actually tried looking at documentation to make sure that `-Wall` and `-Wextra` do not imply `-Wundef`, and I ran some experiments to check my conclusion. But I had never heard of `-Wundef-prefix`, and it didn't occur to me that I would need to experiment with a name that starts `TARGET_OS_`. Thanks again, Apple.